src/main/java/cms/gov/madie/measure/services/MeasureService.java

```diff
@@ -76,6 +76,16 @@ public Group createOrUpdateGroup(Group group, String measureId, String username)
     return group;
   }
 
+  public Measure deleteMeasureGroup(String measureId, String groupId) {
+    Measure measure = measureRepository.findById(measureId).orElse(null);
+
+    List<Group> remainingGroups =
+        measure.getGroups().stream().filter(g -> !g.getId().equals(groupId)).toList();
```

Comment:
You need to do a bit more to address @nmorasb comments

Trying adding if (measure == null) {} to wrap any code that calls measure.getGroups so that an NPE won't be thrown if measure is null.

Reply:
This is outdated code Greg, I have updated it after Nate comments